packages/devtools_app/lib/src/debugger/flutter/codeview.dart

```diff
@@ -82,140 +117,208 @@ class _CodeViewState extends State<CodeView> {
 
     gutterController.dispose();
     textController.dispose();
+
+    widget.controller.scriptLocation
+        .removeListener(_handleScriptLocationChanged);
   }
 
-  void _updateLines() {
-    setState(() {
-      lines = widget.script?.source?.split('\n') ?? [];
+  void _handleScriptLocationChanged() {
+    _updateScrollPosition();
+  }
 
-      executableLines = {};
+  void _updateScrollPosition({bool animate = true}) {
+    if (widget.controller.scriptLocation.value.scriptRef != scriptRef) {
+      return;
+    }
 
-      // TODO(devoncarew): Improve this with SourceReportRange.possibleBreakpoints.
-      // (see getSourceReport('PossibleBreakpoints').
-      // Recalculate the executable lines.
-      if (widget.script?.tokenPosTable != null) {
-        for (var encodedInfo in widget.script.tokenPosTable) {
-          executableLines.add(encodedInfo[0]);
-        }
-      }
-    });
-  }
+    final location = widget.controller.scriptLocation.value?.location;
+    if (location?.line == null) {
+      return;
+    }
+
+    if (!verticalController.hasAttachedControllers) {
+      // TODO(devoncarew): I'm uncertain why this occurs.
```

Comment:
is this code path getting triggered after the stateful widget is disposed?

Reply:
change to log from print or remove.

Reply:
changed to `log()`

Reply:
I added a guard on one of the code paths here (`if (mounted)`); I left the check log code to see if this still fires.